### PR TITLE
Correct RegionTrans.validate to always work on list of nodes (closes #952)

### LINKED ISSUE #952: RegionTrans.validate() crashes if passed a single Node
The docstring for `RegionTrans` says it acts on either a single node or a list of nodes. However, if we attempt to apply the `ACCDataTrans` subclass to a single node then the validate method fails with:

      File "/home/kbc59144/Projects/PSyclone/src/psyclone/psyir/transformations/region_trans.py", line 149, in validate
     node_parent = node_list[0].parent
     TypeError: 'NoneType' object is not subscriptable

### PR DESCRIPTION
**Pull request by arporter**
Small bug fix that uses the existing `RegionTrans.get_node_list()` at the beginning of the `validate` method so as to ensure we're always working with a list of nodes.

While looking at the tests, I discovered we had an old and duplicated test in transformations_test.py so I've removed that.

### CODE REVIEW

**Comment by arporter** [comment]
Ready for first review (assuming Travis is happy).

**Review by rupertford — CHANGES_REQUESTED** [review]
Just one small suggestion. Please see inline comment.

File: src/psyclone/psyir/transformations/region_trans.py [hunk 1]
```diff
@@ -113,9 +115,9 @@ def validate(self, node_list, options=None):
         '''Checks that the nodes in node_list are valid for a region
         transformation.
 
-        :param node_list: list of PSyIR nodes or a single Schedule.
-        :type node_list: :py:class:`psyclone.psyir.nodes.Schedule` or a \
-                         list of :py:class:`psyclone.psyir.nodes.Node`
+        :param node_list: list of PSyIR nodes or a single node.
```
[the remark this answers is not in the document]
Comment (arporter) [reply src/psyclone/psyir/transformations/region_trans.py:118]: Yes, that is better.

**Review by rupertford — APPROVED** [review]
The one small issue has been addressed.

**Pull request merged**